RAC-4064 - common/api_utils need to use LOG and new config

larry-dean opened the pull request:
* Changed all prints to LOG
* Use fitargs instead of system environment
* Added exception handling of API() calls
* Variable names are now consistent
* api_node_select_argList() to api_node_select_from_config()

larry-dean commented:
@stuart-stanley @hohene @johren @jimturnquist

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+
+class NodesTests(unittest.TestCase):
+
+    def setUp(self):
```
Is this code a converted CIT script, or is it meant to test api_utils? It feels a bit like the former with the use of CIT things, but, well the name :). 

I find it very hard to tell what each one is _doing_ in any case. "test_all_list" can mean just about anything..
You can use a one-line doc-string to give more detailed info like (for test_all_list):
    "test performing api_utils.nodes_get_all results in same list shows.... what IS this showing?"

larry-dean replied:
I'll add comment blocks

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+        node_list = self.__get_data()
+
+        id_list = api_node_select(self.__client)
+        self.assertNotEqual(len(node_list), len(id_list), msg='Id list not equal to node list')
```
please add the values of the failed comparison into the assert (like you do in test_type_list()

larry-dean replied:
The function of this test is duplicated by test_type_list(). I'm going to remove test_all_list().

larry-dean commented on test/common/api_utils.py:
```diff
@@ -155,29 +155,32 @@ def api_validate_node_pollers(client, node_id_list, all_pollers=False):
-            print '**** Unable to retrieve node: {} pollers via API. status ({})\n'.format(node.get('id'),
-                                                                                           node_poller_rsp.status)
+    for node_id in node_id_list:
+        #rsp = Api().nodes_get_pollers_by_id(node_id, callback=None)
```
Oops...

larry-dean commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+"""
+Copyright 2016, EMC, Inc.
+
+Author(s):
```
ok

larry-dean commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+import unittest
+
+#CIT
+from modules.logger import Log
```
ok

larry-dean replied:
removed

larry-dean commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+
+from common.api_utils import api_node_select_from_config, api_node_select, api_validate_node_pollers
+
+LOG = Log(__name__)
```
ok

larry-dean replied:
removed

larry-dean commented:
Push changes addressing the following:
1) removed concept of 'all' and replaced with None. Added __fitargs_get() to help with fitargs stuff.
2) reformatted logs() messages

larry-dean commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+        self.assertFalse(polled_response, msg='Got True for bad poller (foo)')
+        
+        
+        
```
removed

larry-dean commented:
I think this push address all of @stuart-stanley  issues

stuart-stanley commented on test/common/api_utils.py:
```diff
@@ -135,12 +143,8 @@ def api_node_select(client,
-            print node_id_list, '\n'
-        else:
-            print '**** Empty node ID list.\n'
+    if not  node_id_list:
```
If we want to keep the logging the same, this would be:
{code}
if node_id_list:
    log.debug_7('Node ID List: Empty node ID List')
else:
    log.debug_7('Node ID List: %s', node_id_list)
{code}

(this may have been changed based on other input. If so, never mind!)

larry-dean replied:
Because these functions are part of the infrastructure, I considered this clutter and removed it.

stuart-stanley commented on test/common/api_utils.py:
```diff
@@ -155,29 +159,30 @@ def api_validate_node_pollers(client, node_id_list, all_pollers=False):
         # if the poller return None or an empty list, return a False
         if not node_poller_list:
-            print 'Node {} has no pollers.'.format(node_ld)
+            logs.debug('Node %s has no pollers.', node_id)
```
Is this debug because this is now a very common state? I'm guessing that's the case, but wanted to check.

larry-dean replied:
This could just be an issue with obm settings. I'll leave it debug.

stuart-stanley commented on test/common/api_utils.py:
```diff
@@ -155,29 +159,30 @@ def api_validate_node_pollers(client, node_id_list, all_pollers=False):
+        except (TypeError, ValueError) as e:
+            assert (e.message)
+        except ApiException as e:
+            logs.debug('Unable to retrieve node: %s pollers via API. status (%s)', node_id, e.status)
```
Should this be warning or error instead of debug? It seem more "serious" than the "no pollers" case below in the sense that it threw an exception...

larry-dean replied:
Your right. This should be iether a error or warning. Will change.

hohene replied:
We should not be making asserts in a library, report the error back to the calling test script.

larry-dean replied:
These assert are for failure of the library not any test.

stuart-stanley commented on test/common/api_utils.py:
```diff
@@ -274,5 +285,4 @@ def get_by_string(source_dict, search_string, default_if_not_found=None):
-    return dict_obj
-        
\ No newline at end of file
+    return dict_obj
```
nit: missing newline (which has to be annoying, since it came from cleaning up the extra spaces on the last line! Sorry!)

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,249 @@
+
+    def test_type_list(self):
+        """
+        Test of the api_node_select() function
```
How does this show up when you run this? I know a one-line doc-string will augment what gets shown in the 'nosetests -v' output. I'm not sure what a multi-line one does?

"validate test utility method api_node_select() output per node_type matches raw nodes_get_all() data"

is what I was aiming for. Sorry I wasn't more clear...

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,249 @@
+        Validate the number of each node type found using nodes_get_all() is the
+        same as the number found be api_node_select().
+
+        This tests the ode_type= parameter.
```
node_type

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,249 @@
+        node_list = self.__get_data()
+        type_count = 0
+
+        print '\n'
```
I'm sorry, I missed this last round. We should be moving to infra-logging for new things.

This would be a straight 'logs.debug('Testing node type %s', type_entry)' I believe.

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,249 @@
+        This validates that an invalid node will not not detect running pollers.
+        """
+        polled_response = api_validate_node_pollers(self.__client, ['foo'], all_pollers=True)
+        self.assertFalse(polled_response, msg='Got True for bad poller (foo)')
```
nit: missing newline

larry-dean commented:
New changes pushed up:
* To prevent inclusion of library log  messages in the console, all api_utiles() messages are now debug.
* logs are changes to logs.irl
* doc strings changed in api_utils_test()

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+            type_count = 0
+
+    def test_node_id_list(self):
+        """Validates test utility method api_node_select() detects if a provided node id is valid"""
```
very very nit (ignore if you want or if you are just tired of me commenting ;)) Since these show up as test-cases, using lowercase and an active voice basically looks better on the nose "validate test utility method blah blah is valid ... ok" output.

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+
+
+    def test_skuName(self):
+        """Validates test utility method api_node_select() outputs all node for a provided sku name"""
```
nit: 'nodes'

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+
+    def test_node_obm_validation(self):
+
+        """Validates test utility method api_node_select() using node id will validated node OBM settings"""
```
nit: 'validate node OBM...' ?

stuart-stanley commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+        self.assertTrue(polled_response, msg='Not all pollers active')
+ 
+    def test_poller_all(self):
+        """Validates test utility method api_validate_node_pollers() will return nodes with all active poller"""
```
nit: pollers

hohene commented on test/common/test/api_utils_test.py:
```diff
@@ -0,0 +1,203 @@
+        return loads(self.__client.last_response.data)
+
+    def test_type_list(self):
+        """validates test utility method api_node_select() using an obm mac address will return the correct node id"""
```
Nit:  nose/unittest doesn't parse the triple ticks when we do a list of the tests in the file.
Add a # to the comment directly under the def test_xxxx and rest of tests below.
(oh...unless you can fix unittest for me.....)  :-)
Otherwise, it displays like this:
```
run_tests.py -stack 13 -test test/api_utils_test.py  -list
*** Created config file: config/generated/fit-config-20170202-000127
*** Using config file: config/generated/fit-config-20170202-000127

Test Listing for: test/api_utils_test.py 
----------------------------------------------------------------------
*** Reloading config file: config/generated/fit-config-20170202-000127
2017-02-02 00:01:28,550 INFO    removing previous logging dir /emc/hohene/sandbox-github/rackhd/stu-rackhd/test/log_output/run_2017-02-01_23:00:10.d 28274 MainProcess stream-monitor/flogging/infra_logging.py:__init__@119 gl-main
2017-02-02 00:01:28,551 INFO    this runs logging dir /emc/hohene/sandbox-github/rackhd/stu-rackhd/test/log_output/run_2017-02-02_00:01:28.d 28274 MainProcess stream-monitor/flogging/infra_logging.py:__init__@119 gl-main
validates test utility method api_node_select_from_config() using obm mac address will return the correct node id ... ok
validates test utility method api_node_select_from_config() will return the correct nodes with valid obm settings ... ok
validates test utility method api_node_select() outputs nodes with validate obm settings ... ok
validates test utility method api_node_select() detects invalid sku type ... ok
validates test utility method api_node_select() detects if a provided node id is valid ... ok
validates test utility method api_node_select() using node id will validate node obm settings ... ok
validates test utility method api_node_select() using obm mac address will return the correct node id ... ok
validates test utility method api_validate_node_pollers() will return nodes with all active pollers ... ok
validates test utility method api_validate_node_pollers() will detect pollers for invalid node ... ok
validates test utility method api_validate_node_pollers() will return nodes with one active poller ... ok
Validates test utility method api_node_select() outputs all nodes for a provided sku name ... ok
validates test utility method api_node_select() using an obm mac address will return the correct node id ... ok

----------------------------------------------------------------------
Ran 12 tests in 0.009s
```

larry-dean replied:
Will comment them out until something changes.

larry-dean commented:
Pushed up code with commented out docstrings.

johren commented:
test this please

larry-dean commented:
test this please

larry-dean commented:
@johren can you push the button on this or do I need to find another core committer?
